src/JsonSerialisation.ts

```diff
@@ -48,16 +41,214 @@ export class JsonSerialisation{
-		let jsonObj: JObject = {};
-
+	public static WriteDictionaryRuntimeObjs(writer: SimpleJson.Writer, dictionary: Map<string, InkObject>){
+		writer.WriteObjectStart();
 		for (let [key, value] of dictionary){
-			let runtimeObj = asOrNull(value, InkObject);
-			if (runtimeObj != null)
-				jsonObj[key] = this.RuntimeObjectToJToken(runtimeObj);
+			writer.WritePropertyStart(key);
+			this.WriteRuntimeObject(writer, value);
+			writer.WritePropertyEnd();
+		}
+		writer.WriteObjectEnd();
+	}
+
+	public static WriteListRuntimeObjs(writer: SimpleJson.Writer, list: InkObject[]){
+		writer.WriteArrayStart();
+		for (let value of list){
+			this.WriteRuntimeObject(writer, value);
+		}
+		writer.WriteArrayEnd();
+	}
+
+	public static WriteIntDictionary(writer: SimpleJson.Writer, dict: Map<string, number>){
+		writer.WriteObjectStart();
+		for (let [key, value] of dict){
+			writer.WriteIntProperty(key, value);
+		}
+		writer.WriteObjectEnd();
+	}
+
+	public static WriteRuntimeObject(writer: SimpleJson.Writer, obj: InkObject): void{
+		// var container = obj as Container;
+		let container = asOrNull(obj, Container);
+		if (container) {
+			return this.WriteRuntimeContainer(writer, container);
+		}
+
+		// var divert = obj as Divert;
+		let divert = asOrNull(obj, Divert);
+		if (divert) {
+			let divTypeKey = '->';
+			if (divert.isExternal)
+				divTypeKey = 'x()';
+			else if (divert.pushesToStack) {
+				if (divert.stackPushType == PushPopType.Function)
+					divTypeKey = 'f()';
+				else if (divert.stackPushType == PushPopType.Tunnel)
+					divTypeKey = '->t->';
+			}
+
+			let targetStr;
+			if (divert.hasVariableTarget)
+				targetStr = divert.variableDivertName;
+			else
+				targetStr = divert.targetPathString;
+
+			writer.WriteObjectStart();
+			writer.WriteProperty(divTypeKey, targetStr);
+
+			if (divert.hasVariableTarget)
+				writer.WriteProperty('var', true);
+
+			if (divert.isConditional)
+				writer.WriteProperty('c', true);
+
+			if (divert.externalArgs > 0)
+				writer.WriteIntProperty('exArgs', divert.externalArgs);
+
+			writer.WriteObjectEnd();
+		}
+
+		// var choicePoint = obj as ChoicePoint;
+		let choicePoint = asOrNull(obj, ChoicePoint);
+		if (choicePoint) {
+			writer.WriteObjectStart();
+			writer.WriteProperty('*', choicePoint.pathStringOnChoice);
+			writer.WriteIntProperty('flg', choicePoint.flags);
+			writer.WriteObjectEnd();
+		}
+
+		// var intVal = obj as IntValue;
+		let intVal = asOrNull(obj, IntValue);
+		if (intVal)
+			writer.WriteInt(intVal.value);
+
+		// var floatVal = obj as FloatValue;
+		let floatVal = asOrNull(obj, FloatValue);
+		if (floatVal)
+			writer.WriteFloat(floatVal.value);
+
+		// var strVal = obj as StringValue;
+		let strVal = asOrNull(obj, StringValue);
+		if (strVal) {
+			if (strVal.isNewline)
+				writer.Write('\n', false);
+			else
+			writer.WriteStringStart();
+			writer.WriteStringInner('^');
+			writer.WriteStringInner(strVal.value);
+			writer.WriteStringEnd();
```

Comment:
We're missing braces around this block

Reply:
Do you want to keep them? I think they're here to _fake_ a hierarchy, but otherwise, I found them a bit confusing.

Reply:
I think they are required, otherwise if `strVal.isNewline` is true we'll still execute `WriteStringInner('^'); WriteStringInner(strVal.value); etc`, even though we shouldn't?

Reply:
Sorry, I was mistaken. I thought it was about another construct somewhere else. You're absolutely right and yes I fixed it in the last commit!